src/main/java/org/prebid/server/bidder/grid/GridBidder.java

```diff
@@ -1,56 +1,243 @@
+                                                String siteKeywords,
+                                                Keywords firstImpExtKeywords,
+                                                Keywords requestExtKeywords) {
+        return GridKeywordsUtil.merge(
+                mapper,
+                GridKeywordsUtil.resolveKeywordsFromOpenRtb(userKeywords, siteKeywords, mapper),
+                GridKeywordsUtil.resolveKeywords(firstImpExtKeywords, mapper),
+                GridKeywordsUtil.resolveKeywords(requestExtKeywords, mapper));
+    }
+
+    private ExtRequest modifyExtRequest(ExtRequest extRequest, Keywords keywords) {
+        final boolean extRequestPresent = extRequest != null;
+        final ExtRequestPrebid extRequestPrebid = extRequestPresent ? extRequest.getPrebid() : null;
+        final Map<String, JsonNode> extRequestProperties = extRequestPresent
+                ? extRequest.getProperties()
+                : Collections.emptyMap();
+
+        final Map<String, JsonNode> modifiedExtRequestProperties =
+                GridKeywordsUtil.modifyWithKeywords(extRequestProperties, keywords, mapper);
+        if (modifiedExtRequestProperties.isEmpty()) {
+            return null;
+        }
+
+        final ExtRequest modifiedBidRequestExt = ExtRequest.of(extRequestPrebid);
+        modifiedBidRequestExt.addProperties(modifiedExtRequestProperties);
+        return modifiedBidRequestExt;
     }
 
     @Override
-    protected BidType getBidType(Bid bid, List<Imp> imps) {
-        final String impId = bid.getImpid();
+    public final Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(httpCall.getRequest().getPayload(), bidResponse), Collections.emptyList());
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+        return bidsFromResponse(bidRequest, bidResponse);
+    }
+
+    private List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> constructBidderBid(bid, bidRequest.getImp(), bidResponse.getCur()))
+                .collect(Collectors.toList());
+    }
+
+    private BidderBid constructBidderBid(Bid bid, List<Imp> imps, String cur) {
```

Comment:
1. Pls rename to `toBidderBid(..)` to be in one style with project code.
2. `cur` is too short.